distributed/core.py

```diff
@@ -693,6 +705,17 @@ def _measure_tick(self):
                 tick_duration,
             )
         self.digest_metric("tick-duration", tick_duration)
+        self._observed_tick_durations.append((now, tick_duration))
+        # TODO: Do we even care that this is accurate? If the event loop is
+        # blocked for long, we'd store data for a longer period of time. The
+        # stuff below just makes it accurate.
+        while (
+            self._observed_tick_durations
+            and now - self._observed_tick_durations[0][0]
+            # called very often!
```

Comment:
This is called once per append, so it should be fine. We canalso refactor the calculation to move everything but the size comparison outside of the loop.

Reply:
> This is called once per append, so it should be fine. We canalso refactor the calculation to move everything but the size comparison outside of the loop.

1. This is just a partial comment. the original was longer... I meant that the measure_tick is called often
2. Still, this while loop is triggered more often than once. If we have a couple of very fast ticks, followed by a very slow tick, we'll pop this multiple times. Worst case is if a tick is longer than `tick.retention` in which case we'd pop the entire deque (except of the last entry). But amortized cost is still constant.